### PR TITLE
CORE-5801 Add membership schemas for validation

### PR DESCRIPTION
**Pull request by charlieR3**
This PR introduces a schema for validating the a group policy is well formed. It needs to support a dynamic group, and static group and work as a group policy for the MGM CPI.

Services to expose this schema and validate against group policy samples will be created later. 

This file has been put together based on agreement in: https://github.com/corda/platform-eng-design/blob/master/core/corda-5/corda-5.0/the-host/group-policy.md
_Note: The samples in the above doc don't all compile after applying agreed updates to schema so this PR resolves those minor issues: https://github.com/corda/platform-eng-design/pull/482_

### CODE REVIEW

File: data/group-schema/src/main/resources/net/corda/schema/group/policy.1.0/corda.group.policy.json [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+          "required": [
+            "members"
+          ],
+          "properties": {
+            "members": {
+              "description": "The member templates for the members to be added to the static network member list.",
+              "type": "array",
+              "items": {
+                "description": "A member template as a map of values for a member to be added to the static network member list.",
+                "type": "object"
```
Comment (dimosr) [inline data/group-schema/src/main/resources/net/corda/schema/group/policy.1.0/corda.group.policy.json]: Should we be imposing a specific structure for each member item here?

**Review by dimosr — COMMENTED** [review]
I had a quick look and it looks quite reasonable to me for a start (I didn't try to think through all the regexp edge cases though!) - just one question. When we later get to validation, we can spend some more time, add tests etc. but I think it shouldn't block us from progressing with other work.

File: data/group-schema/src/main/resources/net/corda/schema/group/policy.1.0/corda.group.policy.json (hunk 1 above)
Comment (charlieR3) [reply data/group-schema/src/main/resources/net/corda/schema/group/policy.1.0/corda.group.policy.json]: I did have that originally, but since we changed to the key-value approach, similar to the member info `LayeredPropertyMap`, I felt it was better to keep this open for any string-string pairs since we will basically just output all of the MGMs member info here which we may add to. Also it would be hard to validate for specific keys, specifically for arrays of data where the keys will contain indexes.

**Comment by charlieR3** [comment]
> When we later get to validation, we can spend some more time, add tests etc. but I think it shouldn't block us from progressing with other work.

Absolutely. We'll include plenty of tests around this when the services consuming this are added.

File: data/group-schema/src/main/resources/net/corda/schema/group/policy.1.0/corda.group.policy.json (hunk 1 above)
Comment (dimosr) [reply data/group-schema/src/main/resources/net/corda/schema/group/policy.1.0/corda.group.policy.json]: ah ok, that makes sense - I wasn't considering the free-form key-value approach will be used here too.

Comment (charlieR3) [reply data/group-schema/src/main/resources/net/corda/schema/group/policy.1.0/corda.group.policy.json]: Ah, sorry. My comment was in regards to MGM Info. I misread where in the schema this comment was. But the answer is essentially the same. We may have indexed values. We should aim to have good documentation supporting this.

**Comment by driessamyn** [comment]
FYI @josephzunigadaly 

This should probably also be integrated with the CPI build tool. Not sure if that's on your backlog, @charlieR3 , or something we should look at, but worth liasing with @josephzunigadaly . Probably not a must for DP2.

**Comment by charlieR3** [comment]
> FYI @josephzunigadaly
> 
> This should probably also be integrated with the CPI build tool. Not sure if that's on your backlog, @charlieR3 , or something we should look at, but worth liasing with @josephzunigadaly . Probably not a must for DP2.

There's nothing currently on our backlog for this, no. Agreed it would be useful to validate this when building the CPI also.

File: data/membership-schema/src/main/kotlin/net/corda/schema/membership/provider/MembershipSchemaException.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package net.corda.schema.membership.provider
+
+/**
+ * Exception thrown when requested config schema is not available.
+ */
+class MembershipSchemaException(msg: String) : RuntimeException(msg)
```
Comment (nikinagy) [inline data/membership-schema/src/main/kotlin/net/corda/schema/membership/provider/MembershipSchemaException.kt]: Why not use CordaRuntimeException?

Comment (charlieR3) [reply data/membership-schema/src/main/kotlin/net/corda/schema/membership/provider/MembershipSchemaException.kt]: No good reason! Will update it.

**Comment by mnesbit** [comment]
It would be good to get this updated and make the schemas available to the CPI validation work

**Comment by charlieR3** [comment]
> It would be good to get this updated and make the schemas available to the CPI validation work

Will ask a team member to get this over the line in my absence.

**Review by nikinagy — DISMISSED** [review]
LGTM

**Review by nikinagy — APPROVED** [review]
LGTM

**Review by dimosr — APPROVED** [review]

**Review by ronanbrowne — APPROVED** [review]
LGTM
